backend/lib/endpoints/users.js

```diff
@@ -190,6 +191,51 @@ async function routes(app) {
       return new User(userData).save();
     },
   );
+
+  app.delete("/", { preValidation: [app.authenticate] }, async (req) => {
+    const { userId } = req;
+    // Insufficient scope to delete users on Auth0
+    // expected any of: delete:users,delete:current_user
+    // Please add any of the needed scopes and uncomment the line below
+    // await Auth0.deleteUser(getCookieToken(req), userId);
+    await Promise.all([
+      User.deleteMany({ _id: userId }),
+      Comment.deleteMany({ "author.id": userId }),
```

Comment:
I was thinking about this line and whether it might leave orphan comments...

Since we can have nested comments, the comments deleted here could be parents of other comments made by a different user. As it stands, this would delete the parent comment (made by the user being deleted), but keep the child comments, leaving them orphan.

Not sure what is expected here, but we could delete those by doing an extra `Comments.deleteMany({parent_id: {'$in': deletedCommentIds}})`

Reply:
Also, this query above will work only for up to 1 level of nesting, otherwise this would need to be recursive...
If we're sticking to 1 level this should be fine